matrix_sdk/src/room/state.rs

```diff
@@ -0,0 +1,69 @@
+use std::ops::Deref;
+
+use crate::{room, RoomType};
+
+/// An enum that represents the state of the room.
+#[derive(Debug, Clone)]
+pub enum State {
```

Comment:
I think this name is a bit of a downgrade from `RoomState` and I wasn't quite happy with `RoomState` either. It will be hard to find or understand what this is about since it only mentions that it represents the state of the room. I think it's better to say something along the lines:

> An enum that abstracts over the different states a room can be in.

I wonder if we can get away with `Room` instead and hide the fact that `Room` exists in the base layer, probably not since we have a deref chain to `Room` from the base layer.

Any ideas for a better name?

Reply:
I used `State` for the lack of a better name. We could turn `matrix_sdk_base::Room` into `BaseRoom` (or maybe `StoreRoom`) to be consistent with the `Client` and `BaseClient` and it would free up `Room`.

Reply:
Ah right, that's a great idea. `BaseRoom` sounds fine.

Reply:
great, will make the change :)